exporters/otlp/otlpmetric/internal/metrictransform/metric.go

```diff
@@ -71,50 +70,78 @@ func toNanos(t time.Time) uint64 {
 	return uint64(t.UnixNano())
 }
 
-// CheckpointSet transforms all records contained in a checkpoint into
+// InstrumentationLibraryReader transforms all records contained in a checkpoint into
 // batched OTLP ResourceMetrics.
-func CheckpointSet(ctx context.Context, exportSelector export.ExportKindSelector, res *resource.Resource, cps export.CheckpointSet, numWorkers uint) ([]*metricpb.ResourceMetrics, error) {
-	records, errc := source(ctx, exportSelector, cps)
-
-	// Start a fixed number of goroutines to transform records.
-	transformed := make(chan result)
-	var wg sync.WaitGroup
-	wg.Add(int(numWorkers))
-	for i := uint(0); i < numWorkers; i++ {
+func InstrumentationLibraryReader(ctx context.Context, exportSelector export.ExportKindSelector, res *resource.Resource, ilmr export.InstrumentationLibraryReader, numWorkers uint) (*metricpb.ResourceMetrics, error) {
+	var ilms []*metricpb.InstrumentationLibraryMetrics
+
+	err := ilmr.ForEach(func(lib instrumentation.Library, mr export.Reader) error {
+
+		records, errc := source(ctx, exportSelector, mr)
+
+		// Start a fixed number of goroutines to transform records.
+		transformed := make(chan result)
+		var wg sync.WaitGroup
+		wg.Add(int(numWorkers))
+		for i := uint(0); i < numWorkers; i++ {
+			go func() {
+				defer wg.Done()
+				transformer(ctx, exportSelector, records, transformed)
+			}()
+		}
 		go func() {
-			defer wg.Done()
-			transformer(ctx, exportSelector, records, transformed)
+			wg.Wait()
+			close(transformed)
 		}()
-	}
-	go func() {
-		wg.Wait()
-		close(transformed)
-	}()
 
-	// Synchronously collect the transformed records and transmit.
-	rms, err := sink(ctx, res, transformed)
-	if err != nil {
+		// Synchronously collect the transformed records and transmit.
+		ms, err := sink(ctx, transformed)
+		if err != nil {
+			return nil
+		}
+
+		// source is complete, check for any errors.
+		if err := <-errc; err != nil {
+			return err
+		}
+		if len(ms) == 0 {
+			return nil
+		}
+
+		ilms = append(ilms, &metricpb.InstrumentationLibraryMetrics{
+			Metrics:   ms,
+			SchemaUrl: lib.SchemaURL,
+			InstrumentationLibrary: &commonpb.InstrumentationLibrary{
+				Name:    lib.Name,
+				Version: lib.Version,
+			},
+		})
+		return nil
+	})
+	if len(ilms) == 0 {
 		return nil, err
 	}
 
-	// source is complete, check for any errors.
-	if err := <-errc; err != nil {
-		return nil, err
+	rms := &metricpb.ResourceMetrics{
+		Resource:                      Resource(res),
+		SchemaUrl:                     res.SchemaURL(),
+		InstrumentationLibraryMetrics: ilms,
 	}
-	return rms, nil
+
+	return rms, err
 }
 
 // source starts a goroutine that sends each one of the Records yielded by
-// the CheckpointSet on the returned chan. Any error encoutered will be sent
+// the Reader on the returned chan. Any error encoutered will be sent
```

Comment:
preexisting typo s/encoutered/encountered/

Reply:
This word was misspelled a bunch in this one file -- fixed.